trieste/models/model_interfaces.py

```diff
@@ -419,6 +420,56 @@ def get_observation_noise(self):
 
         return noise_variance
 
+    def optimize(self, dataset: Dataset) -> None:
+        """
+        Optimize the model with the specified `dataset`.
+
+        If any of the kernel's trainable parameters have priors, then we begin model optimization
+        from the best of a random sample from these parameters' priors. For trainable parameters
+        without priors, we begin optimization from their initial values.
+
+        :param dataset: The data with which to optimize the `model`.
+        """
+
+        num_trainable_params_with_priors = tf.reduce_sum(
+            [tf.size(param) for param in self.model.kernel.trainable_parameters if param.prior]
+        )
+
+        if num_trainable_params_with_priors >= 1:  # Find a promising kernel initialization
+            num_random_samples = tf.minimum(1000, 100 * num_trainable_params_with_priors)
```

Comment:
I can see the motivation for increasing the n_inits, but it does risk becoming prohibitively slow if there are many params.